### PR TITLE
Adding transformations for swagger data types

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR includes:
- Transforms for array types in the swagger
- Transforms for dictionary types in the swagger
- Adding the operation group name to response types

### CODE REVIEW

File: src/namer/namer.ts [hunk 1]
```diff
@@ -78,6 +78,7 @@ async function process(session: Session<CodeModel>) {
   // pascal-case and capitzalize acronym operation groups and their operations
   for (const group of values(model.operationGroups)) {
     const details = <Language>group.language.go;
+    const opGroupName = capitalizeAcronyms(pascalCase(group.$key));
```
Comment (jhendrixMSFT) [inline src/namer/namer.ts]: What is `group.$key`?

File: src/namer/namer.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
-/*---------------------------------------------------------------------------------------------
+/*  ---------------------------------------------------------------------------------------------
```
Comment (jhendrixMSFT) [inline src/namer/namer.ts]: Are all these formatting changes from a tool?

File: src/namer/namer.ts (hunk 1 above)
Comment (catalinaperalta) [reply src/namer/namer.ts]: $key contains the operationGroup identifier

image

File: src/namer/namer.ts (hunk 2 above)
Comment (catalinaperalta) [reply src/namer/namer.ts]: I was playing around with the linter. I'll take it away for now. ESLint rules said that there should be a space or tab after /*

Comment (catalinaperalta) [reply src/namer/namer.ts]: It's not all formatted completely according to the linter rules

Comment (catalinaperalta) [reply src/namer/namer.ts]: image

Actually I'm thinking of leaving it unless you have any objections

Comment (jhendrixMSFT) [reply src/namer/namer.ts]: I don't however if this is what the linter wants we should enable it and fix up all the source files to conform (can happen in a separate PR).

Comment (catalinaperalta) [reply src/namer/namer.ts]: Agreed, there are definitely a few more things that the linter has brought up. I'll add those changes to another PR.

**Pull request merged**